flang/test/Fir/real.fir

```diff
@@ -16,7 +16,7 @@ func @bar(%a : f16, %b : f32, %c : f64, %d : f80, %e : f128) -> f80 {
   // CHECK: fptrunc fp128 %{{.*}} to x86_fp80
   %5 = fir.convert %e : (f128) -> f80
   // CHECK-NEXT: call x86_fp80
-  %6 = call @foop(%1, %2, %3, %4, %5) : (f80, f80, f80, f80, f80) -> f80
+  %6 = fir.call @foop(%1, %2, %3, %4, %5) : (f80, f80, f80, f80, f80) -> f80
```

Comment:
This test was failing without this change. Is an investigation necessary to find the cause? The builtin call getting removed while running tco.

Reply:
It's strange to me. What was the error message?

Reply:
Now that I look at it closely, an inlining happened with call op. See the results with the call and fir.call ops below.

with the call op.
```
define x86_fp80 @bar(half %0, float %1, double %2, x86_fp80 %3, fp128 %4) !dbg !3 {
  %6 = fpext half %0 to x86_fp80, !dbg !7
  %7 = fpext float %1 to x86_fp80, !dbg !9
  %8 = fpext double %2 to x86_fp80, !dbg !10
  %9 = fptrunc fp128 %4 to x86_fp80, !dbg !11
  %10 = fadd x86_fp80 %6, %7, !dbg !12
  %11 = fmul x86_fp80 %10, %8, !dbg !14
  %12 = fsub x86_fp80 %11, %3, !dbg !15
  %13 = fdiv x86_fp80 %12, %9, !dbg !16
  %14 = frem x86_fp80 %13, %6, !dbg !17
  ret x86_fp80 %14, !dbg !18
}

define x86_fp80 @foop(x86_fp80 %0, x86_fp80 %1, x86_fp80 %2, x86_fp80 %3, x86_fp80 %4) !dbg !19 {
  %6 = fadd x86_fp80 %0, %1, !dbg !20
  %7 = fmul x86_fp80 %6, %2, !dbg !22
  %8 = fsub x86_fp80 %7, %3, !dbg !23
  %9 = fdiv x86_fp80 %8, %4, !dbg !24
  %10 = frem x86_fp80 %9, %0, !dbg !25
  ret x86_fp80 %10, !dbg !26
}
```

with the fir.call op
```
define x86_fp80 @bar(half %0, float %1, double %2, x86_fp80 %3, fp128 %4) !dbg !3 {
  %6 = fpext half %0 to x86_fp80, !dbg !7
  %7 = fpext float %1 to x86_fp80, !dbg !9
  %8 = fpext double %2 to x86_fp80, !dbg !10
  %9 = fptrunc fp128 %4 to x86_fp80, !dbg !11
  %10 = call x86_fp80 @foop(x86_fp80 %6, x86_fp80 %7, x86_fp80 %8, x86_fp80 %3, x86_fp80 %9), !dbg !12
  ret x86_fp80 %10, !dbg !13
}

define x86_fp80 @foop(x86_fp80 %0, x86_fp80 %1, x86_fp80 %2, x86_fp80 %3, x86_fp80 %4) !dbg !14 {
  %6 = fadd x86_fp80 %0, %1, !dbg !15
  %7 = fmul x86_fp80 %6, %2, !dbg !17
  %8 = fsub x86_fp80 %7, %3, !dbg !18
  %9 = fdiv x86_fp80 %8, %4, !dbg !19
  %10 = frem x86_fp80 %9, %0, !dbg !20
  ret x86_fp80 %10, !dbg !21
}
```